packages/qdrant-loader/src/qdrant_loader/core/qdrant_manager.py

```
@@ -213,32 +282,50 @@ def create_collection(self) -> None:
-                        if embeddings_cfg is not None
-                        else None
+                vs = self.settings.llm_settings.embeddings.vector_size
+            except AttributeError:
+                vs = None
+            if vs is not None:
+                try:
+                    vector_size = int(vs)
+                except (TypeError, ValueError) as exc:
+                    raise ValueError(
+                        "Invalid global.llm.embeddings.vector_size; expected an integer"
+                    ) from exc
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)